src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -169,7 +227,9 @@ winrt::com_ptr<winrt::Microsoft::Terminal::Settings::Model::implementation::Prof
 // - true iff the json object has the same `GUID` as we do.
 bool Profile::ShouldBeLayered(const Json::Value& json) const
 {
-    if (!_Guid.has_value())
+    // guid was not set
+    const auto myGuid{ Guid() };
+    if (myGuid == winrt::guid{})

```

Comment:
I'm pretty sure there's a difference between not having a GUID and having the default-constructed GUID as a value. IIRC, the default-constructed one is all zeros, and that _is_ a valid GUID setting!